router/core/header_rule_engine.go

```diff
@@ -99,7 +113,21 @@ func (h HeaderRuleEngine) OnOriginRequest(request *http.Request, ctx RequestCont
 					// Headers are case-insensitive, but Go canonicalize them
 					// Issue: https://github.com/golang/go/issues/37834
 					if regex.MatchString(name) {
-						request.Header.Set(name, ctx.Request().Header.Get(name))
+						// Rename the header when matching is provided
+						if rule.Rename != "" && rule.Named == "" {
+							value := ctx.Request().Header.Get(name)
+							if value != "" {
+								request.Header.Set(rule.Rename, ctx.Request().Header.Get(name))
+								request.Header.Del(name)
+							} else if rule.Default != "" {
```

Comment:
Good catch! This will fix `default` in the regex case.

Reply:
thanks